src/Compilers/Core/Portable/SourceGeneration/Nodes/TransformNode.cs

```diff
@@ -73,7 +75,22 @@ public NodeStateTable<TOutput> UpdateStateTable(DriverStateTable.Builder builder
                 {
                     var stopwatch = SharedStopwatch.StartNew();
                     // generate the new entries
-                    var newOutputs = _func(entry.Item, cancellationToken);
+                    ImmutableArray<TOutput> newOutputs;
+                    if (_wrapUserFunc)
+                    {
+                        try
+                        {
+                            newOutputs = _func(entry.Item, cancellationToken);
+                        }
+                        catch (Exception e) when (!ExceptionUtilities.IsCurrentOperationBeingCancelled(e, cancellationToken))
+                        {
+                            throw new UserFunctionException(e);
+                        }
+                    }
+                    else
```

Comment:
We could get rid of the if/else and edit the `when` on the catch so we don't catch if `!_wrapUserFunc` 

I'd then renamed `_wrapUserFunc` to just `CatchAnalyzerExceptions`

Reply:
Went ahead and made the modification to the when clause. 

I can be convinced otherwise, but I think I'd prefer to keep the current name (_wrapUserFunc) as the existing _func name doesn't convey a relationship to analyzers.

Reply:
Generators are analyzers too, we sort of use the terms interchangeably inside the compiler, but I also don't have strong feelings on the naming.